### PR TITLE
Paint fixes (dylwhich's version)

### PR DESCRIPTION
**Pull request by dylwhich**
### Description

All of Bryce's fixes rebased onto current main, plus I got rid of the green plus in the image title that shouldn't have made it into the final version.

### Test Instructions

Use paint, particularly the menu and saving and things, make sure there's no bugs.

### Readiness Checklist
- [x] I have run `make format` to format the changes
- [x] I have compiled the firmware and the changes have no warnings
- [x] I have compiled the emulator and the changes have no warnings
- [x] I have run `make cppcheck` and checked that `cppcheck_result.txt` has no warnings for the changes
- [x] I have added doxygen comments to any code used by more than one Swadge mode. This includes `/*! \file` comments with Design Philosophy, Usage, and Example sections for new headers.
- [x] I have run `make docs` and checked that `doxy_warnings.txt` has no warnings for the new code

### CODE REVIEW

File: main/modes/mfpaint/mode_paint.c [hunk 1]
```diff
@@ -205,6 +204,12 @@ void paintMenuInitialize(void)
 
 void paintSetupMainMenu(void)
 {
+    if (paintMenu->menu != NULL)
+    {
+        deinitMenu(paintMenu->menu);
+    }
```
Comment (AEFeinstein) [inline main/modes/mfpaint/mode_paint.c:210]: `deinitMenu()` doesn't set `paint->menu` to `NULL`, so maybe it's possible to double-free here? I didn't check the rest of the code paths.

File: main/modes/mfpaint/paint_browser.c [hunk 2]
```diff
@@ -644,3 +656,65 @@ void imageBrowserButton(imageBrowser_t* browser, const buttonEvt_t* evt)
...
+        {
+            return ARRAY_SIZE(deleteStr);
+        }
+
+        default:
+        {
+            return 0;
+        }
+    }
+}
```
Comment (AEFeinstein) [inline main/modes/mfpaint/paint_browser.c:720]: Could this function be replaced with `strlen()`? Also why is `BROWSER_DELETE` different than the others?

File: main/modes/mfpaint/paint_draw.c [hunk 3]
```diff
@@ -75,19 +78,24 @@ static const char dialogOutOfSpaceTitleStr[] = "Save Failed!";
...
+
+const char dialogOptionCancelStr[]      = "Cancel";
+static const char dialogOptionSaveStr[] = "Save";
+const char dialogOptionSaveAsStr[]      = "Save as...";
+static const char dialogOptionExitStr[] = "Quit";
+static const char dialogOptionOkStr[]   = "OK";
+
+const uint8_t toolWheelLoadStrLen      = ARRAY_SIZE(toolWheelLoadStr);
+const uint8_t dialogOptionCancelStrLen = ARRAY_SIZE(dialogOptionCancelStr);
+const uint8_t dialogOptionSaveAsStrLen = ARRAY_SIZE(dialogOptionSaveAsStr);
```
Comment (AEFeinstein) [inline main/modes/mfpaint/paint_draw.c:98]: Again, `strlen()` exists...

**Review by AEFeinstein — CHANGES_REQUESTED** [review]

**Comment by AEFeinstein** [comment]
Note to self, when this gets merged, also delete https://github.com/AEFeinstein/Swadge-IDF-5.0/tree/paint-fixes

File: main/modes/mfpaint/mode_paint.c (hunk 1 above)
Comment (dylwhich) [reply main/modes/mfpaint/mode_paint.c]: I think this should be covered by the other paths, but no reason not to add it anyway.

**Review by AEFeinstein — APPROVED** [review]

**Comment by dylwhich** [comment]
I think that's everything? And apologies for the negative `hideControls` bool, but doing it that way means I only have to change it in one place instead of all the other places since it gets `calloc()`'d...

**Comment by AEFeinstein** [comment]
Everything as far as I can tell!

**Pull request merged**